chapter07/src/main/java/com/seaofnodes/simple/node/RegionNode.java

```diff
@@ -71,5 +74,14 @@ private int findDeadInput() {
         if( lhs==null ) return null;
         _idepth = lhs._idepth+1;
         return (_idom=lhs);
-    } 
+    }
+
+    boolean isFinished() {
```

Comment:
yeah, I find the `isFinished` ugly also.  For a non-worklist variant SoN (Simple, right now), it suffices to *not call peephole* until we're done.  As soon as we get a worklist, we'll need something like this.  Right now in AA, I use "is_keep" and look for a special use-flag (the KeepNode uses Nodes I want to keep around).  Elsewhere in Simple we use a `null` use to indicate "I am not yet dead!".

Reply:
The problem with not calling peephole is that an `add(phi, 0)` will idealize to the `phi` and then call the peephole. So there still needs to be some way to stop that.